plugins/inputs/sqlserver/sqlserver.go

```diff
@@ -345,6 +350,36 @@ func (s *SQLServer) Stop() {
 	}
 }
 
+func (s *SQLServer) gatherServerWithRetry(pool *sql.DB, query Query, acc telegraf.Accumulator, connectionString string) error {
+	for retryAttempt := 0; retryAttempt <= s.RetryCount; retryAttempt++ {
+		queryError := s.gatherServer(pool, query, acc, connectionString)
+		if queryError == nil {
+			// gatherServer completed successfully
+			break
+		}
+
+		if retryAttempt >= s.RetryCount {
+			// gatherServer failed on the final retry attempt
+			return queryError
+		}
+
+		log.Printf("W! [inputs.sqlserver] Error gathering data, retrying %d more time(s): %s.", s.RetryCount-retryAttempt, queryError)
```

Comment:
We should get rid of `log` and use the telegraf logging facitlity.
```suggestion
		s.Log.Warnf("Error gathering data, retrying %d more time(s): %s.", s.RetryCount-retryAttempt, queryError)
```